### PR TITLE
feat(RESTPatchAPIGuildMember): add modify current member and deprecate nick route

### LINKED ISSUE #261: Route documentation inconsistent for "Modify Current Member"
**Please describe the problem you are having in as much detail as possible:**
Seeing other functions like [guildVoiceState](https://github.com/discordjs/discord-api-types/blob/315ce3584917635b93a26123470f37a10bd8d846/rest/v9/index.ts#L736) and `user`
They have a default value of "@me" and the type is `Snowflake | @me` and the routes are mentioned
```ts
       /**
	 * Route for:
	 * - PATCH `/guilds/{guild.id}/voice-states/@me`
	 * - PATCH `/guilds/{guild.id}/voice-states/{user.id}`
	 */
	guildVoiceState(guildId: Snowflake, userId: Snowflake | '@me' = '@me') {
		return `/guilds/${guildId}/voice-states/${userId}` as const;
	},
 ```
However for the [`guildMember`](https://github.com/discordjs/discord-api-types/blob/315ce3584917635b93a26123470f37a10bd8d846/rest/v9/index.ts#L239) function, the type  for `userId` is just Snowflake, it doesn't have a default value and the [`@me`](https://discord.com/developers/docs/resources/guild#modify-current-member) is completely undocumented
shouldn't this be consistent?

### PR DESCRIPTION
**Pull request by BenjammingKirby**
**Please describe the changes this PR makes and why it should be merged:**
adds support for the [Modify Current Member](https://discord.com/developers/docs/resources/guild#modify-guild-member) to the guildMember function
also adds a deprecation warning for [Modify Current User Nick](https://discord.com/developers/docs/resources/guild#modify-current-user-nick)

fixes #261 
**Reference Discord API Docs PRs or commits:**

### CODE REVIEW

**Comment by BenjammingKirby** [comment]
I also only added the deprecation warnings for the v9 files, I wasn't sure if they should be added for v8 as well

**Comment by vladfrangu** [comment]
They should be added to v8 too

**Review by kyranet — APPROVED** [review]

File: deno/rest/v8/guild.ts [hunk 1]
```diff
@@ -448,6 +448,8 @@ export interface RESTPatchAPICurrentGuildMemberNicknameJSONBody {
 
 /**
  * https://discord.com/developers/docs/resources/guild#modify-current-user-nick
+ *
+ * @deprecated Use [Modify Current Member](https://discord.com/developers/docs/resources/guild#modify-current-member) instead.
```
Comment (vladfrangu) [inline deno/rest/v8/guild.ts:452]: Could you deprecate RESTPatchAPICurrentGuildMemberNicknameJSONBody too? and also add `RESTPatchAPICurrentGuildMemberJSONBody`?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: rest/v8/guild.ts [hunk 2]
```diff
@@ -445,9 +447,19 @@ export interface RESTPatchAPICurrentGuildMemberNicknameJSONBody {
 	 */
 	nick?: string | null;
 }
+export interface RESTPatchAPICurrentGuildMemberJSONBody {
```
Comment (vladfrangu) [inline rest/v8/guild.ts:450]: Spacing

File: rest/v9/guild.ts [hunk 3]
```diff
@@ -451,9 +453,18 @@ export interface RESTPatchAPICurrentGuildMemberNicknameJSONBody {
 	 */
 	nick?: string | null;
 }
-
+export interface RESTPatchAPICurrentGuildMemberJSONBody {
```
Comment (vladfrangu) [inline rest/v9/guild.ts:456]: Spacing

**Review by vladfrangu — CHANGES_REQUESTED** [review]
Please add new lines where I reviewed 🙏

**Comment by BenjammingKirby** [comment]
> Please add new lines where I reviewed 🙏

done, also I forgot to link to the documentation on the new interfaces apparently, that too has been added

**Review by vladfrangu — APPROVED** [review]

File: deno/rest/v9/mod.ts [hunk 4]
```diff
@@ -259,6 +260,7 @@ export const Routes = {
 	/**
 	 * Route for:
 	 * - PATCH `/guilds/{guild.id}/members/@me/nick`
+	 * @deprecated Use guildMember instead.
```
Comment (suneettipirneni) [inline deno/rest/v9/mod.ts:263]:
```suggestion
	 * @deprecated Use {@link Routes.guildMember} instead.
```

**Review by suneettipirneni — CHANGES_REQUESTED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

File: deno/rest/v8/guild.ts [hunk 5]
```diff
@@ -446,8 +448,22 @@ export interface RESTPatchAPICurrentGuildMemberNicknameJSONBody {
 	nick?: string | null;
 }
 
+/**
+ * https://discord.com/developers/docs/resources/guild#modify-current-member
+ */
+export interface RESTPatchAPICurrentGuildMemberJSONBody {
```
Comment (vladfrangu) [inline deno/rest/v8/guild.ts:454]: This PR needs to be rebased to account for the changes in rest done to support TS 4.5!

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/rest/v8/guild.ts (hunk 5 above)
Comment (BenjammingKirby) [reply deno/rest/v8/guild.ts:454]: so what needs to be changed? just make it use the `AddUndefinedToPossiblyUndefinedPropertiesOfInterface`?

Comment (vladfrangu) [reply deno/rest/v8/guild.ts:454]: Yep

Comment (BenjammingKirby) [reply deno/rest/v8/guild.ts]: done

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**